Umweltzone/Umweltzone
# Update project setup (SDK 30, AGP 7.0.3, Java 11, Kotlin 1.5.31)

# Description
- Use Gradle wrapper v.7.2.
- Use targetSdkVersion 30 and compileSdkVersion 30.
- Use Android Gradle Plugin v.7.0.3, compile against Java 11.
- Use play-services-maps v.17.0.1.
- Drop support for Android 4.3.1 Jelly Bean (API 18) and below.
- Use kotlin v.1.5.31.
- Use jackson 2.13.0.
- Use R8 instead of ProGuard.

# Test subjects
- Fresh installation
- Installation update
- Check share app feature
- Check lock screen feature
- Unit tests
- Instrumentation tests (on Google Pixel 2 device, Android 11)

# Successfully tested `release` build on
- Google Pixel 2 device, Android 11 (API 30)
- Emulator, Android 4.4.2 (API 19) with Google Play Services
